### PR TITLE
EVG-14818: add admin settings for pods in AWS

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-14857

### Description 
Create admin settings specific to pods in AWS that use Secrets Manager and ECS.

### Testing 
Standard admin settings tests. I also tested it manually in local evergreen to verify that the admin page works as expected.

### CODE REVIEW

File: rest/model/admin.go [hunk 1]
```diff
@@ -1496,9 +1473,206 @@ func (a *APIAWSConfig) ToService() (interface{}, error) {
...
+		return nil, nil
+	}
+	return evergreen.S3Credentials{
+		Key:    utility.FromStringPtr(a.Key),
+		Secret: utility.FromStringPtr(a.Secret),
+		Bucket: utility.FromStringPtr(a.Bucket),
+	}, nil
+}
+
+type APIAWSPodConfig struct {
```
Comment (julianedwards) [inline rest/model/admin.go:1518]: Documentation

File: rest/model/admin.go [hunk 2]
```diff
@@ -1496,9 +1473,206 @@ func (a *APIAWSConfig) ToService() (interface{}, error) {
...
+		return nil
+	default:
+		return errors.Errorf("%T is not a supported type", h)
+	}
+}
+
+func (a *APIECSConfig) ToService() (interface{}, error) {
+	if a == nil {
+		return nil, nil
+	}
```
Comment (julianedwards) [inline rest/model/admin.go:1604]: Nit: add new line after this.

File: rest/model/admin.go [hunk 3]
```diff
@@ -1496,9 +1473,206 @@ func (a *APIAWSConfig) ToService() (interface{}, error) {
...
+// Manager.
+type APISecretsManagerConfig struct {
+	SecretPrefix *string `json:"secret_prefix"`
+	Region       *string `json:"region"`
+}
+
+func (a *APISecretsManagerConfig) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case evergreen.SecretsManagerConfig:
+		a.SecretPrefix = utility.ToStringPtr(v.SecretPrefix)
```
Comment (julianedwards) [inline rest/model/admin.go:1660]: Should we also copy the region?

File: rest/model/admin.go [hunk 4]
```diff
@@ -1496,9 +1473,206 @@ func (a *APIAWSConfig) ToService() (interface{}, error) {
...
+		return errors.Errorf("%T is not a supported type", h)
+	}
+}
+
+func (a *APISecretsManagerConfig) ToService() (interface{}, error) {
+	if a == nil {
+		return nil, nil
+	}
+	return evergreen.SecretsManagerConfig{
+		SecretPrefix: utility.FromStringPtr(a.SecretPrefix),
```
Comment (julianedwards) [inline rest/model/admin.go:1672]: Region?

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: rest/model/admin.go (hunk 1 above)
Comment (Kimchelly) [reply rest/model/admin.go:1518]: Added.

File: rest/model/admin.go (hunk 2 above)
Comment (Kimchelly) [reply rest/model/admin.go:1604]: Done.

File: rest/model/admin.go (hunk 3 above)
Comment (Kimchelly) [reply rest/model/admin.go:1660]: Sorry, I forgot to remove this. I deleted the region (which isn't used).

File: rest/model/admin.go (hunk 4 above)
Comment (Kimchelly) [reply rest/model/admin.go:1672]: Same as above. Removed it.

**Review by julianedwards — APPROVED** [review]

**Pull request merged**